packages/techdocs-common/src/stages/publish/azureBlobStorage.ts

```diff
@@ -133,75 +137,101 @@ export class AzureBlobStoragePublish implements PublisherBase {
    * Directory structure used in the container is - entityNamespace/entityKind/entityName/index.html
    */
   async publish({ entity, directory }: PublishRequest): Promise<void> {
+    // First, retrieve a list of all individual files in currently existing
+    const remoteFolder = getCloudPathForLocalPath(entity);
+    let existingFiles: string[];
     try {
-      // Note: Azure Blob Storage manages creation of parent directories if they do not exist.
-      // So collecting path of only the files is good enough.
-      const allFilesToUpload = await getFileTreeRecursively(directory);
-
-      // Bound the number of concurrent batches. We want a bit of concurrency for
-      // performance reasons, but not so much that we starve the connection pool
-      // or start thrashing.
-      const limiter = limiterFactory(BATCH_CONCURRENCY);
-
-      const promises = allFilesToUpload.map(sourceFilePath => {
-        // Remove the absolute path prefix of the source directory
-        // Path of all files to upload, relative to the root of the source directory
-        // e.g. ['index.html', 'sub-page/index.html', 'assets/images/favicon.png']
-        const relativeFilePath = path.normalize(
-          path.relative(directory, sourceFilePath),
-        );
+      existingFiles = await this.getAllBlobsFromContainer({
+        prefix: remoteFolder,
+        maxPageSize: BATCH_CONCURRENCY,
+      });
+    } catch (e) {
+      const errorMessage = `Unable to list file(s) to Azure. ${e}`;
+      this.logger.error(errorMessage);
+      throw new Error(errorMessage);
```

Comment:
Wonder if we should just log the error here, but let the publish attempt continue?  Not super familiar with Azure permissions, but other providers break out `get` and `list` permissions, and we probably don't want to fail the entire publish command just for this.